clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java

```diff
@@ -634,12 +634,12 @@ public boolean matches(ClientRequest request) {
 
 
     @Test
-    public void testExcludeInternalTopicsConfigOption() { 
+    public void testExcludeInternalTopicsConfigOption() {
         subscriptions.subscribe(Pattern.compile(".*"), rebalanceListener);
 
         metadata.update(TestUtils.singletonCluster(TopicConstants.GROUP_METADATA_TOPIC_NAME, 2), time.milliseconds());
 
-        assertFalse(subscriptions.partitionAssignmentNeeded());
+        assertTrue(subscriptions.partitionAssignmentNeeded());
```

Comment:
You are right about number of partition changes being caught. One problem I noticed with my sample consumer is there seems to be a slight delay for consumer assignment to be refreshed once consumer subscription changes. It looks like with the scenarios I checked just refreshing the metadata should be enough. I'll update the patch. Thanks for your feedback.